core/internal/gethwrappers/versions.go

```diff
@@ -109,7 +108,11 @@ func WriteVersionsDB(db *IntegratedVersion) error {
 	if err != nil {
 		return errors.Wrapf(err, "while opening %s", versionsDBPath)
 	}
-	defer logger.ErrorIfCalling(f.Close)
+	defer func() {
+		if cerr := f.Close(); cerr != nil {
+			err = multierr.Append(err, cerr)
+		}
+	}()
```

Comment:
Chose not to inject based on similar funcs following this pattern and this seems like it could be a real problem.